gatsby/create-pages/create-blog-post-overview-pages.ts

```diff
@@ -0,0 +1,64 @@
+import { CreatePagesArgs, PageProps } from 'gatsby';
+import path from 'path';
+
+const BLOG_OVERVIEW_TEMPLATE_PATH = path.resolve(
+  `${process.cwd()}/src/templates/blog-post-overview.tsx`,
+);
+
+export interface BlogOverviewPageContext {
+  limit: number;
+  skip: number;
+  numPages: number;
+  currentPage: number;
+}
+
+export const createBlogPostOverviewPages = async ({
+  actions,
+  reporter,
+  graphql,
+}: CreatePagesArgs) => {
+  const { createPage } = actions;
+
+  const contentfulBlogPages = await graphql<{
+    allContentfulBlogPost: {
+      nodes: {
+        fields: {
+          path: string;
+        };
+      }[];
+    };
+  }>(`
+    {
+      allContentfulBlogPost(sort: { fields: publicationDate, order: DESC }) {
+        nodes {
+          fields {
+            path
+          }
+        }
+      }
+    }
+  `);
+
+  if (contentfulBlogPages.errors || !contentfulBlogPages.data) {
+    reporter.panicOnBuild(
+      `Error while running GraphQL query for blog pagination pages.`,
+    );
+    return;
+  }
+
+  const posts = contentfulBlogPages.data.allContentfulBlogPost.nodes;
+  const postsPerPage = 10;
+  const numPages = Math.ceil(posts.length / postsPerPage);
```

Comment:
Like `postsPerPage` you can also use the full name here: `numberOfPages`. It's just simpler to read.